panic when invalid credentials are used

**Describe the bug**

```
 no provider specified, defaulting to local.
  Use --help for a list of available providers.
→ using ssh authentication method, generate temporary credentials
→ using service account credentials
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
→ connecting to asset spacerocket.fritz.box (baremetal)
x failed to synchronize asset to Mondoo Platform  error="rpc error: code = PermissionDenied desc = no valid permissions to access the resource"

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x10a01ac64]

goroutine 1 [running]:
go.mondoo.com/cnspec/cli/reporter.(*defaultReporter).printSummary(0x1400169fb00, {0x140005a2660, 0x1, 0x1?})
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/cli/reporter/print_compact.go:85 +0xc04
go.mondoo.com/cnspec/cli/reporter.(*defaultReporter).print(0x1400169fb00)
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/cli/reporter/print_compact.go:70 +0x274
go.mondoo.com/cnspec/cli/reporter.(*Reporter).Print(0x10a061af7?, 0x10a05a895?, {0x10ca42d00?, 0x14000010018?})
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/cli/reporter/reporter.go:56 +0xe4
go.mondoo.com/cnspec/apps/cnspec/cmd.printReports(0x1400074e5a0?, 0x1400074e5a0, 0x0?)
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/apps/cnspec/cmd/scan.go:581 +0x124
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func19(0x1400012d620?, {0x11095e278?, 0x5b?, 0x0?}, 0x0?, 0x0?)
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/apps/cnspec/cmd/scan.go:362 +0xf8
go.mondoo.com/cnquery/apps/cnquery/cmd/builder.NewProviderCommand.func2(0x140016cac00?, {0x11095e278?, 0x0, 0x0})
	/Users/chris/go/src/go.mondoo.com/workspace/cnquery/apps/cnquery/cmd/builder/builder.go:60 +0x110
github.com/spf13/cobra.(*Command).execute(0x140016cac00, {0x11095e278, 0x0, 0x0})
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x5b0
github.com/spf13/cobra.(*Command).ExecuteC(0x110764780)
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x35c
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/chris/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/apps/cnspec/cmd/root.go:76 +0x28
main.main()
	/Users/chris/go/src/go.mondoo.com/workspace/cnspec/apps/cnspec/cnspec.go:6 +0x1c
```

**To Reproduce**

- use invalid credentials
- run scan

**Expected behavior**

There is a proper erorr message for the asset

**Screenshots or CLI Output**
If applicable, add screenshots or the CLI ouput to help explain your problem.

**Desktop (please complete the following information):**
- OS: [e.g. macOS]
- OS Version: [e.g. 13.0]
- Browser if applicable [e.g. Chrome, Firefox]
- Browser Version: [e.g. 106]

**Additional context**
Add any other context about the problem here.

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -60,7 +60,10 @@
       "args": [
         "scan",
         "local",
-        "--incognito"
+        "-o",
+        "summary",
+        "--config",
+        "/tmp/mondoo.yml"
       ],
       "env": {
       }

--- a/cli/reporter/print_compact.go
+++ b/cli/reporter/print_compact.go
@@ -81,13 +81,19 @@ func (r *defaultReporter) printSummary(orderedAssets []assetMrnName) {
 		if asset.Url != "" {
 			assetUrl = asset.Url
 		}
-		assetsByPlatform[asset.PlatformName] = append(assetsByPlatform[asset.PlatformName], asset)
-		assetScore := r.data.Reports[assetMrn].Score.Rating().Letter()
-		assetsByScore[assetScore]++
+		if asset.PlatformName != "" {
+			assetsByPlatform[asset.PlatformName] = append(assetsByPlatform[asset.PlatformName], asset)
+		}
+		if _, ok := r.data.Reports[assetMrn]; ok {
+			assetScore := r.data.Reports[assetMrn].Score.Rating().Letter()
+			assetsByScore[assetScore]++
+		}
 	}
 
-	header := fmt.Sprintf("Scanned %d assets", len(r.data.Assets))
-	r.out.Write([]byte(termenv.String(header + NewLineCharacter).Foreground(r.Colors.Primary).String()))
+	if len(assetsByScore) > 0 {
+		header := fmt.Sprintf("Scanned %d assets", len(r.data.Assets))
+		r.out.Write([]byte(termenv.String(header + NewLineCharacter).Foreground(r.Colors.Primary).String()))
+	}
 
 	// print assets by platform
 	r.printAssetsByPlatform(assetsByPlatform)
@@ -154,7 +160,9 @@ func (r *defaultReporter) printSummary(orderedAssets []assetMrnName) {
 			r.out.Write([]byte("Detailed information is already available via the web UI: "))
 			r.out.Write([]byte(spaceUrl + NewLineCharacter))
 		}
-		r.out.Write([]byte("To get more information on the CLI, please run this scan with \"-o full\"." + NewLineCharacter))
+		if len(assetsByScore) > 0 {
+			r.out.Write([]byte("To get more information on the CLI, please run this scan with \"-o full\"." + NewLineCharacter))
+		}
 	}
 }
 

